app/models/report_generators/system_performance/fy2018/measure_four.rb

```diff
@@ -357,29 +373,28 @@ def add_stayer_income universe_of_stayers
             universe_of_stayers[index][:latest_non_earned_income] = nil
           end
         end
-        if earliest_group.present?
-          universe_of_stayers[index][:earliest_income] = 0
-          universe_of_stayers[index][:earliest_earned_income] = 0
-          universe_of_stayers[index][:earliest_non_earned_income] = 0
-          if earliest_group[:TotalMonthlyIncome].present?
-            universe_of_stayers[index][:earliest_income] = earliest_group[:TotalMonthlyIncome]
-            universe_of_stayers[index][:earliest_earned_income] = earliest_group[:EarnedAmount] || 0
-            universe_of_stayers[index][:earliest_non_earned_income] = universe_of_stayers[index][:earliest_income] - universe_of_stayers[index][:earliest_earned_income]
-          elsif (amounts = earliest_group.values_at(amount_columns).compact).any?
-            universe_of_stayers[index][:earliest_income] = amounts.sum
-            universe_of_stayers[index][:earliest_earned_income] = earliest_group[:EarnedAmount] || 0
-            universe_of_stayers[index][:earliest_non_earned_income] = universe_of_stayers[index][:earliest_income] - universe_of_stayers[index][:earliest_earned_income]
-          elsif earliest_group[:IncomeFromAnySource] == 99 || earliest_group[:IncomeFromAnySource] = 8 || earliest_group[:IncomeFromAnySource] = 9
-            universe_of_stayers[index][:earliest_income] = nil
-            universe_of_stayers[index][:earliest_earned_income] = nil
-            universe_of_stayers[index][:earliest_non_earned_income] = nil
-          end
+        next unless earliest_group.present?
+
+        universe_of_stayers[index][:earliest_income] = 0
+        universe_of_stayers[index][:earliest_earned_income] = 0
+        universe_of_stayers[index][:earliest_non_earned_income] = 0
+        if earliest_group[:TotalMonthlyIncome].present?
+          universe_of_stayers[index][:earliest_income] = earliest_group[:TotalMonthlyIncome]
+          universe_of_stayers[index][:earliest_earned_income] = earliest_group[:EarnedAmount] || 0
+          universe_of_stayers[index][:earliest_non_earned_income] = universe_of_stayers[index][:earliest_income] - universe_of_stayers[index][:earliest_earned_income]
+        elsif (amounts = earliest_group.values_at(amount_columns).compact).any?
+          universe_of_stayers[index][:earliest_income] = amounts.sum
+          universe_of_stayers[index][:earliest_earned_income] = earliest_group[:EarnedAmount] || 0
+          universe_of_stayers[index][:earliest_non_earned_income] = universe_of_stayers[index][:earliest_income] - universe_of_stayers[index][:earliest_earned_income]
+        elsif earliest_group[:IncomeFromAnySource] == 99 || earliest_group[:IncomeFromAnySource] = 8 || earliest_group[:IncomeFromAnySource] = 9
```

Comment:
Probably intended comparison, not assignment
```suggestion
        elsif earliest_group[:IncomeFromAnySource] == 99 || earliest_group[:IncomeFromAnySource] == 8 || earliest_group[:IncomeFromAnySource] == 9
```

Reply:
This is also dead code.